src/unix/kqueue.c

```diff
@@ -104,8 +109,12 @@ void uv__io_poll(uv_loop_t* loop, int timeout) {
     assert(w->fd >= 0);
     assert(w->fd < (int) loop->nwatchers);
 
-    if ((w->events & POLLIN) == 0 && (w->pevents & POLLIN) != 0) {
-      filter = EVFILT_READ;
+    if (((w->events & POLLIN) == 0 && (w->pevents & POLLIN) != 0) ||
+        ((w->events & UV__POLLPRI) == 0 && (w->pevents & UV__POLLPRI) != 0)) {
+      if ((w->events & POLLIN) == 0 && (w->pevents & POLLIN) != 0)
+        filter = EVFILT_READ;
+      if ((w->events & UV__POLLPRI) == 0 && (w->pevents & UV__POLLPRI) != 0)
+        filter = EV_OOBAND;
```

Comment:
Why handling `UV__POLLPRI` in the same conditional as `POLLIN`? Why don't do it separately as with `POLLOUT`?